express-api/src/services/buildings/buildingSchema.ts

```diff
@@ -0,0 +1,18 @@
+import z from 'zod';
+
+export const BuildingFilterSchema = z.object({
+  pid: z.number().nonnegative().optional(),
```

Comment:
These z.number() calls should all be z.coerce.number(). This is because when you go to use this on the query params in the controller, all the numeric values you are trying to parse will be strings (ie. "23" instead of 23).